### ISSUE TITLE
ci: failure in `wallet_assumeutxo.py --descriptors`

### ISSUE THREAD

**Issue by fanquake** [bitcoin/bitcoin#29234]
Merged in #28838. Seems to be failing on all MSVC CI jobs.

https://github.com/bitcoin/bitcoin/actions/runs/7492122146/job/20394956531?pr=29232#step:27:7862:
```bash
  test  2024-01-11T17:24:30.783000Z TestFramework (INFO): Backup can't be loaded during background sync 
 test  2024-01-11T17:24:30.783000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "D:\a\bitcoin\bitcoin\test\functional\test_framework\util.py", line 140, in try_rpc
                                       fun(*args, **kwds)
                                     File "D:\a\bitcoin\bitcoin\test\functional\test_framework\coverage.py", line 50, in __call__
                                       return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "D:\a\bitcoin\bitcoin\test\functional\test_framework\authproxy.py", line 129, in __call__
                                       raise JSONRPCException(response['error'], status)
                                   test_framework.authproxy.JSONRPCException: remove_all: The process cannot access the file because it is being used by another process.: "D:\a\_temp\test_runner_?_??_20240111_171305\wallet_assumeutxo_47\node1\regtest\w" (-1)
                                   During handling of the above exception, another exception occurred:
                                   Traceback (most recent call last):
                                     File "D:\a\bitcoin\bitcoin\test\functional\test_framework\test_framework.py", line 131, in main
                                       self.run_test()
                                     File "D:\a\bitcoin\bitcoin/test/functional/wallet_assumeutxo.py", line 127, in run_test
                                       assert_raises_rpc_error(-4, "Wallet loading failed. Error loading wallet. Wallet requires blocks to be downloaded, and software does not currently support loading wallets while blocks are being downloaded out of order when using assumeutxo snapshots. Wallet should be able to load successfully after node sync reaches height 299", n1.restorewallet, "w", "backup_w.dat")
                                     File "D:\a\bitcoin\bitcoin\test\functional\test_framework\util.py", line 131, in assert_raises_rpc_error
                                       assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "D:\a\bitcoin\bitcoin\test\functional\test_framework\util.py", line 144, in try_rpc
                                       raise AssertionError("Unexpected JSONRPC error code %i" % e.error["code"])
                                   AssertionError: Unexpected JSONRPC error code -1
```

**Comment by glozow** [comment]
https://github.com/bitcoin/bitcoin/actions/runs/7500571397/job/20419528825?pr=29179

**Comment by glozow** [comment]
https://github.com/bitcoin/bitcoin/actions/runs/7503027363/job/20426899651

**Comment by m3dwards** [comment]
Trying to replicate this locally and at the moment I get a different failure: 

```shell
assert_raises_rpc_error(-4, "Wallet loading failed. Error loading wallet. Wallet requires blocks to be downloaded, and software does not currently support loading wallets while blocks are being downloaded out of order when using assumeutxo snapshots. Wallet should be able to load successfully after node sync reaches height 299", n1.restorewallet, "w", "backup_w.dat")
```

**Comment by achow101** [comment]
> Could this be a bug in the wallet?

Indeed, this looks more like a windows related bug when there's a wallet restoring failure.

> Why would it call `remove_all` on the wallet file of a completely different node?

I don't think it is. I think the problem is more along the lines of the same bitcoind has the database open at the time it tries to cleanup the failed restore.

**Comment by achow101** [comment]
This actually revealed quite a subtle bug in the way that we handle cleaning up after an `AttachChain` failure. I've opened a fix in #29243

**Comment by cecuabin** [comment]
(cfdm^+1/2e+lt1)* = dtt^2 +1/2-4/3.15.dmb
GPI + K3.105^n = R^2*.rsc
dt= 2/3 -icsg8 . pi^2*.int
gu=dd1t1cs^2d2-DT9t'*.lk
Ka = rc^muv^sh2-pu^2/Eccp^8''*.zip
W1^3/2 + d1t1 . w^3/2 = $LUNC + target 300dsh ($0.023)
RB + MAF = Eo - d1t1 . wrs^2|/^o,313
